lib/src/idx/trees/mtree.rs

```diff
@@ -916,37 +1271,28 @@ impl NodeVectors for LeafNode {
 		self.len()
 	}
 
-	fn get_vector(&self, i: usize) -> Result<Arc<Vector>, Error> {
-		self.get_index(i).ok_or(Error::Unreachable).map(|(v, _)| v.clone())
-	}
-
-	fn distribute_entries(
-		mut self,
-		distances: &[Vec<f64>],
-		p1: usize,
-		p2: usize,
-	) -> Result<(Self, f64, Self, f64), Error> {
-		let mut leaf1 = LeafNode::new();
-		let mut leaf2 = LeafNode::new();
-		let (mut r1, mut r2) = (0f64, 0f64);
-		for (i, (v, mut p)) in self.drain(..).enumerate() {
-			let dist_p1 = distances[i][p1];
-			let dist_p2 = distances[i][p2];
-			if dist_p1 <= dist_p2 {
-				p.parent_dist = dist_p1;
-				leaf1.insert(v, p);
-				if dist_p1 > r1 {
-					r1 = dist_p1;
-				}
-			} else {
-				p.parent_dist = dist_p2;
-				leaf2.insert(v, p);
-				if dist_p2 > r2 {
-					r2 = dist_p2;
-				}
+	fn get_objects(&self) -> Vec<Arc<Vector>> {
+		self.keys().cloned().collect()
+	}
+
+	fn extract_node(
+		&mut self,
+		distances: &DistanceCache,
+		p: Arc<Vector>,
+		a: Vec<Arc<Vector>>,
+	) -> Result<(Self, f64, Arc<Vector>), Error> {
+		let mut n = LeafNode::new();
+		let mut r = 0f64;
+		for o in a {
+			let mut props = self.remove(&o).ok_or(Error::Unreachable)?;
+			let dist = *distances.0.get(&(o.clone(), p.clone())).unwrap_or(&0f64);
+			if dist > r {
+				r = dist;
 			}
+			props.parent_dist = dist;
+			n.insert(o, props);
 		}
-		Ok((leaf1, r1, leaf2, r2))
+		Ok((n, r, p))
```

Comment:
Same comment as https://github.com/surrealdb/surrealdb/pull/2738/files#r1352101679 (Just resolve this thread if it doesn't make sense!)

Reply:
See https://github.com/surrealdb/surrealdb/pull/2738#discussion_r1352700769